lib/Controller/SignalingController.php

```diff
@@ -216,17 +216,20 @@ public function getWelcomeMessage(int $serverId): DataResponse {
 				],
 			]);
 
-			if (!$this->signalingManager->isCompatibleSignalingServer($response)) {
-				return new DataResponse(['error' => 'UPDATE_REQUIRED'], Http::STATUS_INTERNAL_SERVER_ERROR);
-			}
-
 			$body = $response->getBody();
-
 			$data = json_decode($body, true);
+
 			if (!is_array($data) || !isset($data['version'])) {
 				return new DataResponse(['error' => 'JSON_INVALID'], Http::STATUS_INTERNAL_SERVER_ERROR);
 			}
 
+			if (!$this->signalingManager->isCompatibleSignalingServer($response)) {
+				return new DataResponse([
+					'error' => 'UPDATE_REQUIRED',
+					'version' => $data['version'] ?? '',
+				], Http::STATUS_INTERNAL_SERVER_ERROR);
+			}
```

Comment:
This block should be moved above the previous `if` block; otherwise, if there is no version, a `JSON_INVALID` message would be returned rather than `UPDATE_REQUIRED` with no version.

Having said that it is unlikely to happen in the real world, though, as the version is included in the response since 0.0.3.

Reply:
Since this is such an old version we really do not care :P

Reply:
lets only do the update message, makes much more sense